nncf/tensorflow/pruning/filter_pruning/algorithm.py

```diff
@@ -240,6 +240,13 @@ def set_pruning_level(self, pruning_level: float,
         if run_batchnorm_adaptation:
             self._run_batchnorm_adaptation()
 
+    @property
+    def maximal_compression_rate(self) -> float:
+        stats = PrunedModelTheoreticalBorderline(
+            self._pruned_layers_num, self._prunable_layers_num, self._min_prunable_flops,
+            self._min_prunable_params, self.full_flops, self.full_params_num)
+        return stats.maximal_pruning_rate
```

Comment:
I suggest to calculate `maximal_pruning_rage` explicitly. Copy paste will be unified in future when common pruning algorithm builder/controller will be introduced.

Reply:
I agree, this is more clean